modules/ansible/ansibleee.go

```diff
@@ -0,0 +1,321 @@
+	// DNSConfig allows to specify custom dnsservers and search domains
+	DNSConfig *corev1.PodDNSConfig `json:"dnsConfig,omitempty"`
+	// Extra vars to be passed to ansible process during execution. This can be used to override default values in plays.
+	ExtraVars map[string]json.RawMessage `json:"extraVars,omitempty"`
+	// Labels - Kubernetes labels to apply to the job
+	Labels map[string]string `json:"labels,omitempty"`
+	// Annotations - Kubernetes annotations to apply to the job
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// Env is a list containing the environment variables to pass to the pod
+	Env []corev1.EnvVar `json:"env,omitempty"`
+}
+
+// EEJobInterface defines the functions required to format AnsibleEE kubernetes jobs
+type EEJobInterface interface {
+	JobForOpenStackAnsibleEE(h *helper.Helper) (*batchv1.Job, error)
+	addEnvFrom(job *batchv1.Job)
+	addMounts(job *batchv1.Job)
+}
+
+// JobForOpenStackAnsibleEE returns a openstackansibleee Job object
+func (a *EEJob) JobForOpenStackAnsibleEE(h *helper.Helper) (*batchv1.Job, error) {
+	ls := labelsForOpenStackAnsibleEE(a.Name, a.Labels)
+
+	args := a.Args
+
+	playbook := a.Playbook
+	if len(args) == 0 {
+		if len(playbook) == 0 {
+			playbook = "playbook.yaml"
+		}
+		args = []string{"ansible-runner", "run", "/runner", "-p", playbook}
+	}
+
+	// ansible runner identifier
+	// if the flag is set we use resource name as an argument
+	// https://ansible-runner.readthedocs.io/en/stable/intro/#artifactdir
+	if !(util.StringInSlice("-i", args) || util.StringInSlice("--ident", args)) {
+		identifier := a.Name
+		args = append(args, []string{"-i", identifier}...)
+	}
+
+	podSpec := corev1.PodSpec{
+		RestartPolicy: corev1.RestartPolicy(a.RestartPolicy),
+		Containers: []corev1.Container{{
+			ImagePullPolicy: "Always",
+			Image:           a.Image,
+			Name:            a.Name,
+			Args:            args,
+			Env:             a.Env,
+		}},
+	}
+
+	if a.DNSConfig != nil {
+		podSpec.DNSConfig = a.DNSConfig
+		podSpec.DNSPolicy = "None"
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        a.Name,
+			Namespace:   a.Namespace,
+			Annotations: a.Annotations,
+			Labels:      ls,
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: a.BackoffLimit,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: a.Annotations,
+					Labels:      ls,
+				},
+				Spec: podSpec,
+			},
+		},
+	}
+
+	// Populate hash
+	hashes := make(map[string]string)
+
+	if len(a.InitContainers) > 0 {
+		job.Spec.Template.Spec.InitContainers = a.InitContainers
+	}
+	if len(a.ServiceAccountName) > 0 {
+		job.Spec.Template.Spec.ServiceAccountName = a.ServiceAccountName
+	}
+	// Set primary inventory if specified as string
+	existingInventoryMounts := ""
+	if len(a.Inventory) > 0 {
+		setRunnerEnvVar(h, "RUNNER_INVENTORY", a.Inventory, "inventory", job, hashes)
+		existingInventoryMounts = "/runner/inventory/inventory.yaml"
```

Comment:
should `/runner/inventory/inventory.yaml`, as well as `playbook.yaml` on L87 be const?

Reply:
Yeah, can probably make these const instead